procurement_auto_create_group_carrier/models/stock_rule.py

```diff
@@ -8,8 +8,8 @@ class StockRule(models.Model):
 
     _inherit = "stock.rule"
 
-    def _prepare_auto_procurement_group_data(self):
-        data = super()._prepare_auto_procurement_group_data()
+    def _prepare_auto_procurement_group_data(self, product_id):
+        data = super()._prepare_auto_procurement_group_data(product_id)
```

Comment:
Based on my comment on https://github.com/OCA/stock-logistics-warehouse/pull/1821:
```suggestion
    def _prepare_auto_procurement_group_data(self, product):
        data = super()._prepare_auto_procurement_group_data(product)
```